hazelcast/src/main/java/com/hazelcast/instance/impl/JetExtension.java

```diff
@@ -49,12 +49,14 @@
     private final ILogger logger;
     private final JetServiceBackend jetServiceBackend;
     private volatile boolean activated;
+    private volatile boolean isAfterStartCalled;
```

Comment:
Have you considered merging this with `activated` field and create some enum? Might be cleaner and more explicit what states are expected and allowed.

Reply:
I cannot properly explain myself (, or name) the state variable that will occur when I combine these two. Maybe they would be combined if i thought differently but I couldn't do it. These two variables seem to me as separate things. I use `activated` to represent state of `Jet`, while I use `isAfterStartCalled` just to synchronize `afterStart` and `onClusterVersionChange` methods.